### DISCUSSION TITLE
How to save data to database by clicking on map in interface

### DISCUSSION THREAD

**Discussion by jglom** [directus/directus#4272] [Archive]
Hello everyone,
currently I am trying to mark the captured latLng values with a marker on the implemented map with an onClick and save these values to the database.

The first part works wonderfully, however saving causes problems as I can't trigger the "Save"-Button with the onClick. However, if I change the fields manually, the "Save"-Button is activated...

How would you solve this? Do you have experience with this?

**Comment by rijkvanzanten** [comment]
> currently I am trying to mark the captured latLng values with a marker on the implemented map with an onClick and save these values to the database.

Instead of just on `input`, you'd want to `emit('input', currentMarker)` whenever you're changing it. It's a little hard to give any further instructions without a more in-depth look into the source code of your extension 🙂 

> Here you can see the map in the dashboard (http://localhost:8055/admin/collections/xyz/+):

The world can't open your localhost :)

**Reply by jglom** [reply]
> Instead of just on input, you'd want to emit('input', currentMarker) whenever you're changing it. It's a little hard to give any further instructions without a more in-depth look into the source code of your extension 🙂

There is actually nothing more to see. I have a map that I initialize where there is an onClick on it. I save the values (lat, lng) in data() and display them in the input fields. Now the "Save"-Button should be activated to save the entry, just as if I enter the field manually.

I am not worried about implementing the request to save the data in the database, but about detecting a change in the input field after clicking on the map and inserting the values.

> The world can't open your localhost :)

this is just an visualization of where the map is located in MY dashboard :)

**Reply by rijkvanzanten** [reply]
> There is actually nothing more to see. I have a map that I initialize where there is an onClick on it. I save the values (lat, lng) in data() and display them in the input fields. Now the "Save"-Button should be activated to save the entry, just as if I enter the field manually.

You say there's nothing more to see, and then proceed to list all other things that are required to be seen 😄 

> but about detecting a change in the input field after clicking on the map and inserting the values.

I'd use a method in Vue that calls emit for you, and instead of relying on `v-model` use that emitter method on input, and base the value attribute on the value prop of the component, that way you keep all in sync and you have a good point to emit that value on both input and click
